src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackTraceExporter.cs

```diff
@@ -243,6 +243,13 @@ internal int SerializeActivity(Activity activity)
             cntFields += 1;
         }
 
+        if (!string.IsNullOrEmpty(activity.TraceStateString))
+        {
+            cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, "traceState");
+            cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, activity.TraceStateString);
```

Comment:
Could this be a non-ASCII string, if that's the case what should be the expected behavior?

Reply:
I understand the spec part (I wrote part of that spec). My question was about the .NET implementation https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.tracestatestring.

Reply:
The .NET implementation accepts non-ASCII characters: [sharplab.io](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAIwCwAUPgAwAE+xALANwXV3EB0AIlBAOYwA9gGcEUAMbCW5VrQBuEZDQA2UUQFMY6pQF4aWgO40AguLFyoCAJ4AZNQk3aKAbwo13NAMoALQamUYdhowACqCNHoA+hEAfDQIaOpwbh6eEGAADsrqETQAFMjqAGYmZlAW1gDChRBigjAA8hl1MMIAPKbmllaV9Q4AHghxkQCUsaVd1mmZqjB8AErqwv4IHMbKyly1EMYwGIvigsgY6hgUAL7SneXdnn7I4uprGBjXFbb2jsh5qsHaI9JWMQqABtAC6NAw21yIIARFQqABaBjAIoATkIRQAzABWADseOAWIYUIAbBAsY8MWiMFR1HSGHisaTEQiilRSXiIBAqMAIGiqIRgHjWcRYXAaLCioJBDoIMBxABLwC8G4AAPdhYMB5B4omQUGAqAcGF6MAGCAACshBBl+LUjjQMlabXw7boaDqEvrDacTWbLdbbQgjhxqupauouMUICt/c7XXkARQKI6Ay6g8gOABRQbICBmEK5x4ASVee08CHDeShFeS5A87FJ+UEwAAVuozAB+GjiRR67QS+i0IpQdQBABy6SSND8CHYVC7w9HGAAahBlKgpzP2Fi2oOYguRwFV+ulmMdDEUu5XHX6x5F0e1xvhLkTkVo8oENJbx4FEpqxBcjyQdwRGHtkD7ZAAEIv2/Gh7xXR8cj0eCJzAHJhAMSxxG8S9b2vWDb1hBI83UG1ClNWFxn/EEqDBWsCPrIjCyWCsHEo89IW2EFiDo3CCOiDjX3fBB6Ngy5cPOUT3C3X94mY81FE0ETcJkxQ5JI8tw0TGRKGINE8mIx5NIcbSKFkvNJisO5UAeJD9HUIw3lue5HjyWEAFkrCcqYXPUWFtPMsp3l9dRBlybyej6ULVgU5BhHUAz5MU01IIlQz1GM9QArUiybmsXJcveazbI4TTkAQCK3M8iLxQmPKrAAaVgDBSu0OR+xoQruhCwZTN0/SuusSCOALDTWIyz05m0oA).

Reply:
I found opentelemetry doing validation for the tracestate's key and value: https://github.com/open-telemetry/opentelemetry-dotnet/blob/4d02fe1a0803daaf6e5db1cf6f45f5d8d89c99a4/src/OpenTelemetry.Api/Context/Propagation/TraceStateUtilsNew.cs#L166

I'll add something similar to validate here as well.

Regarding the expected behavior, [Section 4.3.3](https://www.w3.org/TR/trace-context/#a-traceparent-is-received) of the spec says:
> The vendor MAY validate the tracestate header. If the tracestate header cannot be parsed the vendor MAY discard the entire header. Invalid tracestate entries MAY also be discarded.

Looks like I can either drop invalid entries or the entire header. I'll keep aligned with the core library to drop the entire header.

Reply:
@xiang17 I don't understand, why making this complex instead of just serializing the Unicode string directly? The exporter is not supposed to validate the tracestate content IMHO.

```suggestion
            cursor = MessagePackSerializer.SerializeUnicodeString(buffer, cursor, activity.TraceStateString);
```

Reply:
I had a misunderstanding on it. Updated it to use Unicode.